src/main/java/org/terasology/logic/health/ChangeMaxHealthAuthoritySystem.java

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.logic.health;
+
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.health.event.ChangeMaxHealthEvent;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.NUIManager;
+import org.terasology.rendering.nui.widgets.UIIconBar;
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class ChangeMaxHealthAuthoritySystem extends BaseComponentSystem {
```

Comment:
ideally, other entities should be notified when such a change took place, so we should send out a notification event (immutable) after changing max health - we can further consider whether or not to let the ui update react to that instead of the change event